apiServer/pxone/utils/torpedoutils.go

```diff
@@ -0,0 +1,164 @@
+		for _, n := range nodes {
+			log.InfoD("reboot node: %s", n.Name)
+			err := tests.Inst().N.RebootNode(n, node.RebootNodeOpts{
+				Force: true,
+				ConnectionOpts: node.ConnectionOpts{
+					Timeout:         defaultCommandTimeout,
+					TimeBeforeRetry: defaultCommandRetry,
+				},
+			})
+			if err != nil {
+				c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+				return
+			}
+		}
+		c.JSON(http.StatusOK, gin.H{"message": "All Nodes successfully rebooted"})
+		return
+	} else if nodename == "random" {
+		nodes := node.GetWorkerNodes()
+		randomNode := nodes[rand.Intn(len(nodes))]
+		err := tests.Inst().N.RebootNode(randomNode, node.RebootNodeOpts{
+			Force: true,
+			ConnectionOpts: node.ConnectionOpts{
+				Timeout:         defaultCommandTimeout,
+				TimeBeforeRetry: defaultCommandRetry,
+			},
+		})
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		} else {
+			c.JSON(http.StatusOK, gin.H{"message": fmt.Sprintf("Randomly selected node %s successfully rebooted", randomNode.Name)})
+		}
+	} else {
+		nodes := node.GetWorkerNodes()
+		for _, n := range nodes {
+			if n.Name == nodename {
+				err := tests.Inst().N.RebootNode(n, node.RebootNodeOpts{
+					Force: true,
+					ConnectionOpts: node.ConnectionOpts{
+						Timeout:         defaultCommandTimeout,
+						TimeBeforeRetry: defaultCommandRetry,
+					},
+				})
+				if err != nil {
+					c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+				} else {
+					c.JSON(http.StatusOK, gin.H{"message": fmt.Sprintf("Node with name %s successfully rebooted", n.Name)})
+				}
+				return
+			}
+		}
+		c.JSON(http.StatusNotFound, gin.H{"error": fmt.Sprint("Node with name %s not found", nodename)})
+	}
+}
+
+// GetStorageNodes : Returns all Storage Node objects in the cluster
+func GetStorageNodes(c *gin.Context) {
+	if !checkTorpedoInit(c) {
+		return
+	}
+	nodes, err := tests.GetStorageNodes()
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"error": err.Error(),
+		})
+	} else {
+		c.JSON(http.StatusOK, gin.H{
+			"nodes": nodes,
+		})
+	}
+}
+
+// GetStorageLessNodes : Returns all Storage less node objects in the cluster
+func GetStorageLessNodes(c *gin.Context) {
+	if !checkTorpedoInit(c) {
+		return
+	}
+	nodes := node.GetStorageLessNodes()
+	c.JSON(http.StatusOK, gin.H{"nodes": nodes})
+}
+
+// CollectSupport : This API collects the support bundle
+func CollectSupport(c *gin.Context) {
+	if !checkTorpedoInit(c) {
+		return
+	}
+	tests.CollectSupport()
+	c.JSON(http.StatusOK, gin.H{"message": "Collection of support bundle done from Torpedo End"})
+}
+
+// ScheduleAppsAndValidate : This API schedules multiple applications on the cluster and validates them
+// context is created as a global context to be accessed later in further tests
+func ScheduleAppsAndValidate(c *gin.Context) {
+	if !checkTorpedoInit(c) {
+		return
+	}
+	context = tests.ScheduleApplications(testName)
+	tests.ValidateApplications(context)
```

Comment:
how are you planning to report the failure in case of error in validations?

Reply:
I will be writing another handler to read from errChan in subsequent PRs here. If you want for now we can remove this Validation from here and bring all of that code all in. What do you suggest ?